test/locks.js

```diff
@@ -39,231 +47,219 @@ describe('Locks', function() {
-    return async.series([ lock, unlock ], function(err) {
-      assert(!err);
-      done();
-    });
-    function lock(callback) {
-      return locks.lock('test', callback);
-    }
-    function unlock(callback) {
-      return locks.unlock('test', callback);
+  it('cleanup', async function() {
+    try {
+      const response = await apos.locks.db.remove({});
+
+      assert(response.result.ok === 1);
+    } catch (e) {
+      assert(false);
     }
   });
 
-  it('should allow two differently-named locks uneventfully', function(done) {
-    let locks = apos.modules['apostrophe-locks'];
-    return async.series([ lock1, lock2, unlock1, unlock2 ], function(err) {
-      assert(!err);
-      done();
-    });
-    function lock1(callback) {
-      return locks.lock('test1', callback);
-    }
-    function unlock1(callback) {
-      return locks.unlock('test1', callback);
-    }
-    function lock2(callback) {
-      return locks.lock('test2', callback);
-    }
-    function unlock2(callback) {
-      return locks.unlock('test2', callback);
+  it('should allow a single lock without contention uneventfully', async function() {
+    const locks = apos.modules['apostrophe-locks'];
+
+    try {
+      await locks.lock('test');
+
+      await locks.unlock('test');
+    } catch (e) {
+      assert(false);
     }
+
+    // function lock(callback) {
+    //   return locks.lock('test', callback);
+    // }
+    // function unlock(callback) {
+    //   return locks.unlock('test', callback);
+    // }
   });
 
-  it('should flunk a second lock by the same module', function(done) {
-    let locks = apos.modules['apostrophe-locks'];
-    return async.series([ lock, lockAgain, unlock, unlockAgain ], function(err) {
-      assert(!err);
-      done();
-    });
-    function lock(callback) {
-      return locks.lock('test', callback);
-    }
-    function lockAgain(callback) {
-      return locks.lock('test', function(err) {
-        // SHOULD fail
-        assert(err);
-        return callback(null);
-      });
+  it('should allow two differently-named locks uneventfully', async function() {
+    const locks = apos.modules['apostrophe-locks'];
+
+    try {
+      await locks.lock('test1');
+      await locks.unlock('test1');
+      await locks.lock('test2');
+      await locks.unlock('test2');
+    } catch (e) {
+      assert(false);
     }
-    function unlock(callback) {
-      return locks.unlock('test', callback);
+  });
+
+  it('should flunk a second lock by the same module', async function() {
+    const locks = apos.modules['apostrophe-locks'];
+
+    await locks.lock('test');
+
+    try {
+      await locks.lock('test');
+    } catch (e) {
+      assert(e);
     }
-    function unlockAgain(callback) {
-      return locks.unlock('test', function(err) {
-        // SHOULD fail
-        assert(err);
-        return callback(null);
-      });
+
+    await locks.unlock('test');
+
+    try {
+      await locks.unlock('test');
```

Comment:
same issue, you need an assert(false) after this to stop the show if the code is allowed to continue past the (intentionally) bad second unlock.